src/hotspot/share/opto/loopopts.cpp

```diff
@@ -2297,6 +2299,11 @@ void PhaseIdealLoop::clone_outer_loop(LoopNode* head, CloneLoopMode mode, IdealL
       clone_outer_loop_helper(old, loop, outer_loop, old_new, wq, this, true);
     }
 
+    for (uint i = 0; i < loop->_body.size(); i++) {
+      Node* old = loop->_body.at(i);
+      clone_outer_loop_helper(old, loop, outer_loop, old_new, wq, this, true);
```

Comment:
While you're at it, could you rename the helper method to something more meaningful?

Reply:
Thanks for reviewing this. I did in the new commit. Does it look good to you?